packages/ens-utils/src/ethregistrar.ts

```diff
@@ -155,3 +179,284 @@ export function buildNFTRefFromENSName(
+*/
+export interface PriceDescription {
+  /* descriptiveTextBeginning references the text that is displayed before the price */
+  descriptiveTextBeginning: string;
+  /* pricePerYear is a string that represents: Price + "/ year" (e.g. "$5.99 / year") */
+  pricePerYearDescription: string;
+  /* descriptiveTextBeginning references the text that is displayed after the price */
+  descriptiveTextEnd: string;
+}
+
+export const getPriceDescription = (
+  registration: Registration,
+  parsedName: DomainName,
+): PriceDescription | null => {
+  const isExpired =
+    registration.primaryStatus === PrimaryRegistrationStatus.Expired;
+  const wasRecentlyReleased =
+    registration.secondaryStatus ===
+    SecondaryRegistrationStatus.RecentlyReleased;
+  const isRegistered =
+    registration.primaryStatus === PrimaryRegistrationStatus.Active;
+
+  if (!(isExpired && wasRecentlyReleased) && isRegistered) return null;
+  const domainBasePrice = AvailableNameTimelessPriceUSD(parsedName);
+
+  if (!domainBasePrice) return null;
+  else {
+    const domainPrice = formattedPrice({
+      price: domainBasePrice,
+      withPrefix: true,
+    });
+    const pricePerYearDescription = `${domainPrice} / year`;
+
+    const premiumEndsIn = premiumPeriodEndsIn(registration)?.relativeTimestamp;
+
+    if (premiumEndsIn) {
+      const premiumEndMessage = premiumEndsIn
+        ? ` Temporary premium ends ${premiumEndsIn}.`
+        : null;
+      const basePriceMessage = domainBasePrice
+        ? " Discounts continuously until dropping to "
+        : null;
+
+      return {
+        pricePerYearDescription,
+        descriptiveTextBeginning:
+          "Recently released." + premiumEndMessage + basePriceMessage,
+        descriptiveTextEnd: ".",
+      };
+    } else {
+      const domainLabelLength = parsedName.labelName.length;
+
+      return domainLabelLength <
+        DOMAIN_HAS_SPECIAL_PRICE_IF_LENGTH_EQUAL_OR_LESS_THAN
+        ? {
+            pricePerYearDescription,
+            descriptiveTextBeginning: `${domainLabelLength}-character names are `,
+            descriptiveTextEnd: " to register.",
+          }
+        : null;
+    }
+  }
+};
+
+export const nameCurrentTemporaryPremium = (
+  registration: Registration,
+): Price | null => {
+  if (registration.expirationTimestamp) {
+    return temporaryPremiumPriceAtTimestamp(
+      now(),
+      registration.expirationTimestamp,
+    );
+  } else {
+    return null;
+  }
+};
+
+/* Interface for premium period end details */
+export interface PremiumPeriodEndsIn {
+  relativeTimestamp: string;
+  timestamp: Timestamp;
+}
+
+/**
+ * Determines if a domain is in its premium period and returns the end timestamp and a human-readable distance to it.
+ * @param domainCard: DomainCard
+ * @returns PremiumPeriodEndsIn | null
```

Comment:
Let's please document what it means for this to return null. I see at least a few cases for this, including:

1. registration is not expired
2. registration is not currently in the premium period